src/backend/vulkan/TextureVk.cpp

```diff
@@ -312,22 +306,24 @@ namespace backend { namespace vulkan {
         return VulkanAspectMask(GetFormat());
     }
 
-    // Helper function to add a texture barrier to a command buffer. This is inefficient because we
-    // should be coalescing barriers as much as possible.
-    void Texture::RecordBarrier(VkCommandBuffer commands,
-                                nxt::TextureUsageBit currentUsage,
-                                nxt::TextureUsageBit targetUsage) const {
+    void Texture::TransitionUsageNow(VkCommandBuffer commands, nxt::TextureUsageBit usage) {
+        // Avoid encoding barriers when it isn't needed.
+        if (mLastUsage == usage) {
```

Comment:
Is a barrier not needed if transitioning from a write usage to a write usage?

Reply:
Done